### PR TITLE
feat: add SLF4J for logging

### PR DESCRIPTION
**Pull request by MasterFlomaster1**
Add SLF4J facade to enhance logging and enable users to select their preferred implementation.

### CODE REVIEW

**Comment by literallyfiro** [comment]
I was just working on adding the SLF4J api but seems like you beat me to it!

Code looks good, but could you also add an implementation (like log4j-slf4j or logback) in the demo module?

File: src/main/java/org/teleight/teleightbots/bot/manager/BotManagerImpl.java [hunk 1]
```diff
@@ -71,7 +73,7 @@ private <T extends TelegramBot> void startProcessor(@NotNull T telegramBot, @Not
                     // Happens when the bot token or the username is invalid.
                     // It can also happen if the webhook was considered invalid by the bot API.
                     if (throwable != null) {
-                        System.out.println("An error occurred while authenticating the bot " + telegramBot.getBotUsername() + ": " + throwable.getMessage());
+                        log.error("An error occurred while authenticating the bot {}: {}", telegramBot.getBotUsername(), throwable.getMessage());
```
Comment (literallyfiro) [inline src/main/java/org/teleight/teleightbots/bot/manager/BotManagerImpl.java:76]: I feel like there should be some kind of fallback logging when it comes to error handling.

If you try running a bot with a misconfigured token or username, without a valid slf4j implementation, the program will just silently exit without telling you what went wrong.

Or we could make the implementation required in order to use the library, even though I do not quite like this idea.

**Comment by MasterFlomaster1** [comment]
Added slf4j-simple as fallback option and a simple logback config for demo module

File: gradle/libs.versions.toml [hunk 2]
```diff
@@ -18,6 +18,8 @@ junit-api = { group = "org.junit.jupiter", name = "junit-jupiter-api", version.r
 junit-engine = { group = "org.junit.jupiter", name = "junit-jupiter-engine", version.ref = "junit-jupiter" }
 javalin = { group = "io.javalin", name = "javalin", version.ref = "javalin"}
 javalin-ssl = { group = "io.javalin.community.ssl", name = "ssl-plugin", version.ref = "javalin-ssl"}
+slf4j-api = { module = "org.slf4j:slf4j-api", version = "2.0.16" }
+slf4j-simple = { module = "org.slf4j:slf4j-simple", version = "2.0.16" }
```
Comment (literallyfiro) [inline gradle/libs.versions.toml:22]: Can you put the slf4j module version in the versions category?

```toml
[versions]
# ...
slf4j = "2.0.16"

[libraries]
# ...
slf4j-api = { module = "org.slf4j:slf4j-api", version.ref = "slf4j" }
slf4j-simple = { module = "org.slf4j:slf4j-simple", version.ref = "slf4j" }
```

File: demo/build.gradle.kts [hunk 3]
```diff
@@ -5,6 +5,12 @@ plugins {
 dependencies {
     implementation(rootProject)
     implementation(libs.jetbrains.annotations)
+
+    implementation("ch.qos.logback:logback-classic:1.5.12")
```
Comment (literallyfiro) [inline demo/build.gradle.kts:9]: You can also put this in the `libs.versions.toml` file

**Review by literallyfiro — CHANGES_REQUESTED** [review]
Just a few nitpicks here and there

**Comment by MasterFlomaster1** [comment]
done, also would be nice to update the README with instructions for setting up logging

**Review by literallyfiro — APPROVED** [review]

**Comment by literallyfiro** [comment]
> done, also would be nice to update the README with instructions for setting up logging

I am already working on the documentation website and I will definitely add this

Thank you!

**Pull request merged**